src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(5);
+        switch (index) {
+            case 0:
+                int radiusA = new Random().nextInt(1,10);

```

Comment:
create just one random per class instance, make it a class field
extract literals like 10 to constant
create a private utility method for generating a geometric property, use random there
make ColorSupplier a class field, same as random, generate color once, before switch statement, then use this local variable

Reply:
not fixed